ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/snmp-settings.j2

```diff
@@ -24,6 +24,24 @@
 | {{ row_contact }} | {{ row_location }} | {{ row_traps_disabled }} | Disabled |
 {%         endif %}
 {%     endif %}
+{%     if snmp_server.engine_ids is arista.avd.defined %}
+
+### SNMP EngineID Configuration
+
+| Type | EngineID (Hex) | IP | Port |
+| ---- | ---------- | -- | ---- |
+{%         if snmp_server.engine_ids.local is arista.avd.defined %}
+| local | {{ snmp_server.engine_ids.local }} | - | - |
+{%         endif %}
+{%         for engine_id in snmp_server.engine_ids.remotes %}
+{%             if engine_id.id is arista.avd.defined and engine_id.address is arista.avd.defined %}
+{%                 set row_id = engine_id.id | arista.avd.default('-') %}
+{%                 set row_address = engine_id.address | arista.avd.default('-') %}
```

Comment:
We don't need to set the lines 38,39 as you have already done the **defined** test for **engine_id.id** and **engine_id.address**, which will make sure they are defined as well as not None. You can directly use those vars in line 41.